### ISSUE TITLE
People can subscribe to closed lists

### ISSUE THREAD

**Issue by mherrb** [sympa-community/sympa#1099]
Someone was able to subscribe via an the web interface to a closed list on our server

Version
-------
Sympa 6.2.48

Installation method
-------------------
Source packages

Expected behavior
-----------------
The robot should genarate an error and refuse the action

Actual behavior
---------------
The mail address was added to the subscriber list. 

Additional information
----------------------
Log file for the bad action attached below. IP address replaced with XXX.XXX.XXX.XXX and subscriber email replaced by spammer@example.com

Note that in the mean time we have also explicitely closed subscribtions to the list, so it's not reproducible on our server for this list anymore.

```
Jan 24 20:03:32 budapest wwsympa[15729]: info main::do_info() [robot laas.fr] [session 66137488787641] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:03:39 budapest wwsympa[15729]: info main::do_info() [robot laas.fr] [session 11343586175042] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:03:56 budapest wwsympa[15729]: info main::do_subscribe() [robot laas.fr] [session 11343586175042] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:04:25 budapest wwsympa[15729]: info main::do_subscribe(spammer@example.com) [robot laas.fr] [session 11343586175042] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:04:30 budapest wwsympa[15729]: info main::do_subscribe(spammer@example.com) [robot laas.fr] [session 11343586175042] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:04:30 budapest wwsympa[15729]: notice Sympa::Spindle::ProcessRequest::_twist() Processing Sympa::Request <action=subscribe;context=rtslam-users@laas.fr;email=spammer@example.com>
Jan 24 20:04:30 budapest wwsympa[15729]: notice Sympa::Spool::store() Sympa::Request <action=subscribe;context=rtslam-users@laas.fr;email=spammer@example.com> is stored into Sympa::Spool::Auth as <1611515070,rtslam-users@laas.fr_282c7d4beaa460b5,spammer@example.com,subscribe>
Jan 24 20:04:30 budapest wwsympa[15729]: notice Sympa::Spindle::ProcessTemplate::_twist() Processing Sympa::Message::Template <rtslam-users@laas.fr.1611515070.15729,7147>; envelope_sender=rtslam-users-owner@laas.fr; message_id=sympa.1611515070.15729.329@laas.fr; recipients=spammer@example.com; sender=sympa@laas.fr; template=request_auth; type=subscribe
Jan 24 20:04:30 budapest wwsympa[15729]: notice Sympa::Spool::Outgoing::store() Message Sympa::Message::Template <rtslam-users@laas.fr.1611515070.15729,7147> is stored into bulk spool as <5.5.1611515070.1611515070.753595.rtslam-users@laas.fr_s,15729,883>
Jan 24 20:04:30 budapest wwsympa[15729]: info Sympa::Spindle::ToAuth::_twist() SUBSCRIBE for Sympa::List <rtslam-users@laas.fr> from nobody, auth requested (0.16 seconds)
Jan 24 20:04:30 budapest wwsympa[15729]: info main::do_info() [robot laas.fr] [session 11343586175042] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:04:31 budapest bulk[6719]: notice Sympa::Spindle::ProcessOutgoing::_twist() Start sending message Sympa::Message <5.5.1611515070.1611515070.753595.rtslam-users@laas.fr_s,15729,883/s> to Sympa::List <rtslam-users@laas.fr> (priority 5) (starting 1 seconds after scheduled expedition date)
Jan 24 20:04:31 budapest bulk[6719]: notice Sympa::Mailer::store() Done sending message Sympa::Message <5.5.1611515070.1611515070.753595.rtslam-users@laas.fr_s,15729,883/s> for Sympa::List <rtslam-users@laas.fr> (priority 5) in 1 seconds since scheduled expedition date
Jan 24 20:07:41 budapest wwsympa[15729]: notice Sympa::Spindle::ProcessTemplate::_twist() Processing Sympa::Message::Template <rtslam-users@laas.fr.1611515261.15729,8087/shelved:>; envelope_sender=rtslam-users-owner@laas.fr; message_id=sympa.1611515261.15729.648@laas.fr; recipients=spammer@example.com; sender=rtslam-users-request@laas.fr; template=welcome; 
Jan 24 20:07:41 budapest wwsympa[15729]: notice Sympa::Spool::Outgoing::store() Message Sympa::Message::Template <rtslam-users@laas.fr.1611515261.15729,8087/shelved:> is stored into bulk spool as <1.5.1611515261.1611515261.281781.rtslam-users@laas.fr_s,15729,9540>
Jan 24 20:07:41 budapest wwsympa[15729]: notice Sympa::List::send_notify_to_owner() No owner defined or all of them use nomail option in list Sympa::List <rtslam-users@laas.fr>; using listmasters as default
Jan 24 20:07:41 budapest wwsympa[15729]: notice Sympa::Spindle::ProcessTemplate::_twist() Processing Sympa::Message::Template <rtslam-users@laas.fr.1611515261.15729,1013>; envelope_sender=rtslam-users-owner@laas.fr; message_id=sympa.1611515261.15729.576@laas.fr; recipients=ARRAY; sender=sympa@laas.fr; template=listowner_notification; type=notice
Jan 24 20:07:41 budapest wwsympa[15729]: notice Sympa::Spool::Outgoing::store() Message Sympa::Message::Template <rtslam-users@laas.fr.1611515261.15729,1013> is stored into bulk spool as <5.5.1611515261.1611515261.322530.rtslam-users@laas.fr_s,15729,497>
Jan 24 20:07:41 budapest wwsympa[15729]: info Sympa::Request::Handler::subscribe::_twist() SUB rtslam-users from spammer@example.com accepted (0.10 seconds, 1 subscribers)
Jan 24 20:07:42 budapest bulk[6719]: notice Sympa::Spindle::ProcessOutgoing::_twist() Start sending message Sympa::Message <1.5.1611515261.1611515261.281781.rtslam-users@laas.fr_s,15729,9540/s> to Sympa::List <rtslam-users@laas.fr> (priority 1) (starting 1 seconds after scheduled expedition date)
Jan 24 20:07:42 budapest bulk[6719]: notice Sympa::Mailer::store() Done sending message Sympa::Message <1.5.1611515261.1611515261.281781.rtslam-users@laas.fr_s,15729,9540/s> for Sympa::List <rtslam-users@laas.fr> (priority 1) in 1 seconds since scheduled expedition date
Jan 24 20:07:42 budapest bulk[6719]: notice Sympa::Spindle::ProcessOutgoing::_twist() Start sending message Sympa::Message <5.5.1611515261.1611515261.322530.rtslam-users@laas.fr_s,15729,497/s> to Sympa::List <rtslam-users@laas.fr> (priority 5) (starting 1 seconds after scheduled expedition date)
Jan 24 20:07:42 budapest bulk[6719]: notice Sympa::Mailer::store() Done sending message Sympa::Message <5.5.1611515261.1611515261.322530.rtslam-users@laas.fr_s,15729,497/s> for Sympa::List <rtslam-users@laas.fr> (priority 5) in 1 seconds since scheduled expedition date
Jan 24 20:08:16 budapest wwsympa[15729]: info main::do_info() [robot laas.fr] [session 43313465665732] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:08:32 budapest wwsympa[15729]: info main::do_d_read(/) [robot laas.fr] [session 43313465665732] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
Jan 24 20:08:32 budapest wwsympa[15729]: err main::#1544 > main::do_d_read#11399 [robot laas.fr] [session 43313465665732] [client XXX.XXX.XXX.XXX] [list rtslam-users] Access denied for  because list is not open
Jan 24 20:08:32 budapest wwsympa[15729]: info main::prevent_visibility_bypass() [robot laas.fr] [session 43313465665732] [client XXX.XXX.XXX.XXX] [list rtslam-users] visibility: List must remain hidden. Returning "home" to prevent visibility bypass
Jan 24 20:08:32 budapest wwsympa[15729]: info main::do_home() [robot laas.fr] [session 43313465665732] [client XXX.XXX.XXX.XXX] [list rtslam-users] 
```

**Comment by ikedas** [comment]
Hi @mherrb ,

Could you please show us these things?
- The `config` file of the list, especially the value of `subscribe` parameter in that file.
- The content of corresponding scenario file.  The name is `subscribe.xxx` (`xxx` is the  value of the parameter above).  About location of this file see "[the manual](https://sympa-community.github.io/manual/customize/basics-scenarios.html#location-of-scenario-file)".

**Comment by Salo15** [comment]
Hello,

I have also noticed this behavior on our server in version 6.2.60.
I would expect that subscribing to a disabled list is not possible.
However, that does not seem to be the case and subscribing to a list is possible even if the list is disabled.

Here is the information you requested for 2 cases where subscribing to a disabled list was possible:

1. example (list name replaced by xyz):
# grep -i subscribe xyz/config | grep -v unsub
subscribe auth_notify
# more /etc/sympa/scenari/subscribe.auth_notify
title.gettext Subscription request confirmed (owner is notified)
# do not authentify nor notify updates
is_subscriber ([listname],[sender]) smtp,dkim,smime,md5 -> do_it
true()smtp,dkim -> request_auth
true()md5,smime -> do_it,notify

2. example:
# grep -i subscribe sabinetest5/config | grep -v unsub
subscribe auth
# more /usr/share/sympa/default/scenari/subscribe.auth
# subscribe.auth
title.gettext subscription request confirmed
!equal([sender],[email]) smtp,dkim,md5,smime -> request_auth([email])
is_subscriber([listname],[email]) smtp,dkim,md5,smime -> do_it
true()smtp,dkim -> request_auth([email])
true()md5,smime -> do_it

Is my assumption correct that no subscribe should be possible with disabled lists?

Kind regards,
Sabine Lorenz

**Comment by Salo15** [comment]
Hi @ikedas ,

Thank you for pointing out the parameter allow_subscribe_if_pending, that was the crucial tip.
I did not know this parameter and it was not set on our server.
After I set this parameter to "off", now the closed lists can't be activated either. (by deactivate I meant closing a list via the "remove list" button, which causes the mailing list to be listed in the list of closed mailing lists).
This is exactly what I was looking for and had not found.
I can well imagine that this was also the problem of mherrb, who opened this ticket.

Kind regards,
Sabine
